tests/test_permissions.py

```diff
@@ -25,17 +25,29 @@
 
 
 # registration is done in setUp
-@require_permission(permissions.OVERSEER)
-async def restricted(context):
-    await context.reply("Restricted command was executed.")
+
+
+# @require_permission(permissions.OVERSEER)
+# async def restricted(context):
+#     await context.reply("Restricted command was executed.")
 
 
 @pytest.fixture
 def Setup_fx(bot_fx):
     """Sets up the test environment"""
     Commands._flush()
     Commands.bot = bot_fx
+
+
+@pytest.fixture
+def restricted_command_fx(async_callable_fx, Setup_fx):
+    # ugly hack to utilize async_callable_fx is ugly
+    @require_permission(permissions.OVERSEER)
+    async def restricted(context):
+        await async_callable_fx(context)
```

Comment:
You should be able to do `restricted = require_permission(permissions.OVERSEER)(async_callable_fx)`.

Reply:
didn't think of that. 👍